src/containers/classrooms/ClassroomsTableContainer.jsx

```diff
@@ -1,82 +1,91 @@
-import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Actions } from 'jumpstate';
-import AstroClassroomsTable from '../../components/astro/AstroClassroomsTable';
+import Paragraph from 'grommet/components/Paragraph';
 
+import ConfirmationDialog from '../../components/common/ConfirmationDialog';
+import DarienClassroomsTable from '../../components/darien/DarienClassroomsTable';
+import AstroClassroomsTableContainer from '../astro/AstroClassroomsTableContainer';
 import {
   CLASSROOMS_INITIAL_STATE, CLASSROOMS_PROPTYPES
 } from '../../ducks/classrooms';
 import {
   ASSIGNMENTS_INITIAL_STATE, ASSIGNMENTS_PROPTYPES
 } from '../../ducks/assignments';
-import { programsMocks } from '../../ducks/programs';
 
 class ClassroomsTableContainer extends React.Component {
   constructor() {
     super();
 
     this.state = {
-      toExport: {
-        assignment: {},
-        classroom: {}
-      }
+      classroomToDelete: null,
+      showConfirmationDialog: false
     };
 
-    this.onExportModalClose = this.onExportModalClose.bind(this);
-    this.showExportModal = this.showExportModal.bind(this);
+    this.closeConfirmationDialog = this.closeConfirmationDialog.bind(this);
+    this.deleteClassroom = this.deleteClassroom.bind(this);
+    this.maybeDeleteClassroom = this.maybeDeleteClassroom.bind(this);
   }
 
-  onExportModalClose() {
-    this.setState({ toExport: { assignment: {}, classroom: {} } });
+  maybeDeleteClassroom(id) {
+    this.setState({ classroomToDelete: id, showConfirmationDialog: true });
+  }
 
-    Actions.caesarExports.showModal();
+  closeConfirmationDialog() {
+    this.setState({ classroomToDelete: null, showConfirmationDialog: false });
   }
 
-  showExportModal(assignment, classroom) {
-    this.setState({ toExport: { assignment, classroom } });
+  deleteClassroom() {
+    if (this.state.classroomToDelete === null) return;
+
+    Actions.deleteClassroom(this.state.classroomToDelete).then((response) => {
+      // TODO: For API optimization, do we want to instead manually remove the classroom
+      // out of local app state instead of making another API call
+      Actions.getClassroomsAndAssignments();
+      this.closeConfirmationDialog();
 
-    Actions.caesarExports.showModal();
-    Actions.getCaesarExport({ assignment, classroom });
+      if (response) {
+        Actions.classrooms.setToastState({ status: 'ok', message: 'Classroom deleted' });
+      }
+    });
   }
 
-  selectClassroom(classroom) {
-    Actions.classrooms.selectClassroom(classroom);
+  determineClassroomsTable() {
+    const ClassroomsTable = this.props.selectedProgram.custom ? DarienClassroomsTable : AstroClassroomsTableContainer;
+    return (
+      <ClassroomsTable
+        assignments={this.props.assignments}
+        assignmentsStatus={this.props.assignmentsStatus}
+        classrooms={this.props.classrooms}
+        match={this.props.match}
+        maybeDeleteClassroom={this.props.maybeDeleteClassroom}
```

Comment:
Unsure: Is this missing `deleteClassroom={this.props.deleteClassroom}` ?

Reply:
`maybeDeleteClassroom={this.maybeDeleteClassroom}` This should open the dialog and `this.deleteClassroom` is the click handler for confirmation. I missed removing the `this.props` part here.